packages/punchcard/lib/appsync/type-system.ts

```diff
@@ -0,0 +1,42 @@
+import { ArrayShape, RecordMembers, RecordShape } from '@punchcard/shape';
+import { FunctionShape } from '@punchcard/shape/lib/function';
+import { TraitImpl } from './trait';
+
+export interface TypeSpec {
+  type: RecordShape<{}, string>;
+  fields: RecordMembers;
+  resolvers: TraitImpl<RecordShape<{}, string>, {}>;
+}
+
+/**
+ * A map from a type's FQN to its field-level resolvers.
+ */
+export type TypeSystem = {
+  [fqn in string]: TypeSpec
+};
+
+export namespace TypeSystem {
```

Comment:
I don't see a record type for GraphQL input objects. This specialized type needs to ban circularity, so no pointer shapes... I believe it also needs to be flat (only containing primitives). Could be wrong on that last statement.

Reply:
Actually, this doesn't apply to AppSync, given that the creation of resolvers is happening through the virtual VTL.